src/BuildingRegistry.Producer.Snapshot.Oslo/Infrastructure/Program.cs

```diff
@@ -1,39 +1,166 @@
+                    "FirstChanceException event raised in {AppDomain}.",
+                    AppDomain.CurrentDomain.FriendlyName);
+
+            AppDomain.CurrentDomain.UnhandledException += (_, eventArgs) =>
+                Log.Fatal((Exception)eventArgs.ExceptionObject, "Encountered a fatal exception, exiting program.");
+
+            Log.Information("Initializing BuildingRegistry.Producer.Snapshot.Oslo");
+
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration((_, configurationBuilder) =>
                 {
-                    Hosting =
-                    {
-                        HttpPort = 6016
-                    },
-                    Logging =
+                    configurationBuilder
+                        .SetBasePath(Directory.GetCurrentDirectory())
+                        .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
+                        .AddJsonFile($"appsettings.{Environment.MachineName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
+                        .AddEnvironmentVariables()
+                        .AddCommandLine(args);
+                })
+                .ConfigureLogging((hostContext, loggingBuilder) =>
+                {
+                    SelfLog.Enable(Console.WriteLine);
+                    Log.Logger = new LoggerConfiguration() //NOSONAR logging configuration is safe
+                        .ReadFrom.Configuration(hostContext.Configuration)
+                        .Enrich.FromLogContext()
+                        .Enrich.WithMachineName()
+                        .Enrich.WithThreadId()
+                        .Enrich.WithEnvironmentUserName()
+                        .Destructure.JsonNetTypes()
+                        .CreateLogger();
+                    loggingBuilder.ClearProviders();
+                    loggingBuilder.AddSerilog(Log.Logger);
+                })
+                .ConfigureServices((hostContext, services) =>
+                {
+                    var healthChecksBuilder = services.AddHealthChecks();
+                    var connectionStrings = hostContext.Configuration
+                        .GetSection("ConnectionStrings")
+                        .GetChildren();
+
+                    foreach (var connectionString in connectionStrings
+                                 .Where(x => !x.Value.Contains("host", StringComparison.OrdinalIgnoreCase)))
                     {
-                        WriteTextToConsole = false,
-                        WriteJsonToConsole = false
-                    },
-                    Runtime =
+                        healthChecksBuilder.AddSqlServer(
+                            connectionString.Value,
+                            name: $"sqlserver-{connectionString.Key.ToLowerInvariant()}",
+                            tags: new[] { "db", "sql", "sqlserver" });
+                    }
+
+                    foreach (var connectionString in connectionStrings
```

Comment:
This is for safety/future reasons? I don't think we connect to postgresql here?

Reply:
We do, see appsettings